src/eval/lang.rs

```diff
@@ -0,0 +1,63 @@
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+use crate::coprocessor::Coprocessor;
+use crate::field::LurkField;
+use crate::store::{Ptr, Store};
+use crate::sym::Sym;
+
+// TODO: Define a trait for the Hash and parameterize on that also.
+#[derive(Debug, Default, Clone)]
+pub struct Lang<'a, F: LurkField> {
+    coprocessors: HashMap<Sym, &'a dyn Coprocessor<F>>,
+}
```

Comment:
Is there a meaning for having this struct manipulate *references*, of a *specific lifetime*?
- do we expect coprocessors to be stored elsewhere, and if so where?
- do we expect to hold references to them for any less than `’static`?

Reply:
This was a consequence of the `dyn` needed for the simplest first-level approach. I've adapted your enum trait sandwich now (will push shortly), which does get around that.